kubespawner/spawner.py

```diff
@@ -3236,44 +3260,27 @@ async def load_user_options(self):
         a final fallback.
 
         KubeSpawner recognizes the option named 'profile' and options named like
-        'profile-option-{profile-slug}-{option-slug}'. These user_options will
+        'profile-option-{profile_slug}--{option_slug}'. These user_options will
         be validated against the spawner's profile_list.
 
         Override in subclasses to support other options.
         """
+        # get an initialized profile list
         profile_list = self.profile_list
         if callable(profile_list):
             profile_list = await maybe_future(profile_list(self))
         profile_list = self._get_initialized_profile_list(profile_list)
 
-        selected_profile = self.user_options.get('profile', None)
-        selected_profile_user_options = dict(self.user_options)
-        if selected_profile:
-            # Remove the 'profile' key so we are left with only selected profile options
-            del selected_profile_user_options['profile']
```

Comment:
Since `user_options` are filtered to `selected_profile_user_options` via `_options_from_form` anyhow, and its not an issue to retain `profile` there - I opted to remove the complexity of making a copy in order to mutate the user options.